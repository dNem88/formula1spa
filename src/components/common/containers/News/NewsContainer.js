import React from 'react'
import styles from './NewsContainer.module.css'
import {useNavigate} from 'react-router-dom'

function NewsContainer({id, path, type, children}) {

    let navigate = useNavigate();

    function clickHandler(e) {
        navigate(`/${path}/${id}`)
    }

    return (
        <div onClick={clickHandler} id={id} className={styles[`main-container-${type}`]}>
            {children}
        </div>
    )
}

export default NewsContainer