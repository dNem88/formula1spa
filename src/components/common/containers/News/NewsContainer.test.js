import React from 'react'
import renderer from 'react-test-renderer'
import NewsContainer from './NewsContainer'
import {MemoryRouter} from 'react-router-dom'


const mockProps = {
    id: 'test-id',
    path: 'news',
    type: 'news',
    children: 'CHILDREN'
}
jest.fn('clickHandler')
describe('Testing NewsContainer Component', () => {
    it('render properly based on props', () => {
        const snapshot = renderer.create(
            <MemoryRouter>
                <NewsContainer {...mockProps}/>
            </MemoryRouter>
        ).toJSON()
        expect(snapshot).toMatchSnapshot()
        expect(snapshot.props.className).toBe('main-container-news')
        expect(snapshot.type).toBe('div')
        expect(snapshot.children[0]).toBe('CHILDREN')
    })
    
})