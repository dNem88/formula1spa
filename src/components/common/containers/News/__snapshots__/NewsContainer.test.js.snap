// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing NewsContainer Component render properly based on props 1`] = `
<div
  className="main-container-news"
  id="test-id"
  onClick={[Function]}
>
  CHILDREN
</div>
`;
